Assets/Xbox Live/GameSave/Scripts/GameSaveUI.cs

```diff
@@ -75,19 +79,17 @@ public void InitializeSaveSystem()
     // Update is called once per frame
     void Update()
     {
-        if (XboxLiveComponent.Instance.User != null && XboxLiveComponent.Instance.User.IsSignedIn && !this.gameSaveHelper.IsInitialized() && !this.initializing)
+        if (XboxLiveUserHelper.Instance.SingleUserModeEnabled
+            && XboxLiveUserHelper.Instance.SingleXboxLiveUser.User != null)
         {
-            this.InitializeSaveSystem();
+            this.XboxLiveUser = XboxLiveUserHelper.Instance.SingleXboxLiveUser;
         }
-    }
 
```

Comment:
all of this logic is repeated too much in the project.  
instead, the code pushed into a helper like this:

this.XboxLiveUser = XboxLiveUserHelper.Instance.GetUser(this.XboxLiveUser);

and it'll do if user isn't set, then do this do this logic and set it.